### ISSUE TITLE
Error converting from SpatRaster to RasterLayer

### ISSUE THREAD

**Issue by ailich** [rspatial/terra#432]
As of terra 1.5.1 conversion from terra SpatRaster to raster RasterLayer results in an error. This was previously working in terra 1.5.0.

```
library(terra)
#terra version 1.5.1
library(raster)
r<- rast(volcano)
r2<- raster(r)
#Error: $ operator is invalid for atomic vectors
```

**Comment by Rapsodia86** [comment]
Works fine for me: 
> library(terra)
terra version 1.5.1
Warning messages:
1: multiple methods tables found for ‘approxNA’ 
2: package ‘terra’ was built under R version 4.1.2 
> library(raster)
Loading required package: sp

Attaching package: ‘raster’

The following object is masked from ‘package:terra’:

    approxNA

Warning message:
package ‘raster’ was built under R version 4.1.2 
> r<- rast(volcano)
> r2<- raster(r)
> r
class       : SpatRaster 
dimensions  : 87, 61, 1  (nrow, ncol, nlyr)
resolution  : 0.01639344, 0.01149425  (x, y)
extent      : 0, 1, 0, 1  (xmin, xmax, ymin, ymax)
coord. ref. :  
source      : memory 
name        : lyr.1 
min value   :    94 
max value   :   195 
> r2
class      : RasterLayer 
dimensions : 87, 61, 5307  (nrow, ncol, ncell)
resolution : 0.01639344, 0.01149425  (x, y)
extent     : 0, 1, 0, 1  (xmin, xmax, ymin, ymax)
crs        : NA 
source     : memory
names      : lyr.1 
values     : 94, 195  (min, max)

**Comment by ailich** [comment]
Hmmn, weird. I just updated a few minutes ago so maybe there's an extra comit or something incorporated even though it's the same version number, or maybe the issue is OS or R version specific. This is my system information:

```
R.version
               _                           
platform       x86_64-w64-mingw32          
arch           x86_64                      
os             mingw32                     
system         x86_64, mingw32             
status                                     
major          4                           
minor          0.4                         
year           2021                        
month          02                          
day            15                          
svn rev        80002                       
language       R                           
version.string R version 4.0.4 (2021-02-15)
nickname       Lost Library Book           

packageVersion("terra")
‘1.5.1’

packageVersion("raster")
‘3.5.6’
```

**Comment by rhijmans** [comment]
That is because of change in the behavior of `sources`. I just pushed a change to the raster package; as that is where this coercion should happen. So if you update to `raster 3.5-7` and then update terra develop again, all should work again.  (I need to make sure that I send a new version of `raster` to CRAN before I send a new version of `terra`)

**Comment by ailich** [comment]
Thanks. Works now with raster 3.5.7

**Comment by ailich** [comment]
@rhijmans the conversion doesn't seem to be working the other way (from RasterLayer to SpatRaster).

```
library(terra)
library(raster)
r2<- raster(volcano)
r<- rast(r2)
# Error in methods::as(x, "SpatRaster") : 
#   no method or default for coercing “RasterLayer” to “SpatRaster”

```

**Comment by rhijmans** [comment]
I moved some more code from terra to raster, so I think you need to install the raster package again. Sorry about that.

**Comment by ailich** [comment]
@rhijmans, layer names are lost when converting from SpaterRaster to RasterLayer

```
library(terra)
library(raster)

packageVersion("raster")
# ‘3.5.7’
packageVersion("terra")
# ‘1.5.1’

r<- rast(volcano)
names(r)<- "test"

r2<- raster(volcano)
names(r2)
# "layer"
```

**Comment by rhijmans** [comment]
I do not think so, at least not in your example

```
r2<- raster(r)
names(r)
# [1] "test"
```

(obviously `r2<- raster(volcano)` creates a new object that is not related to `r`)

**Comment by ailich** [comment]
@rhijmans, sorry messed up when pasting into github. I still get the same issue, but maybe the latest version hasn't hit r-universe yet. I'll try again in a few hours.

```
library(terra)
library(raster)

packageVersion("raster")
# ‘3.5.7’
packageVersion("terra")
# ‘1.5.1’

r<- rast(volcano)
names(r)<- "test"

r2<- raster(r)
names(r2)
# "layer"
```

**Comment by rhijmans** [comment]
Ah, I did
 
`names(r)`

in stead of  

`names(r2)`

😫

I will fix this in the `raster` package

**Comment by rhijmans** [comment]
Fixed in `raster 3.5.8`

**Comment by ailich** [comment]
Thanks!
